[docs] Define backport criteria (#125701)
NixOS/nixpkgs: doc/contributing/submitting-changes.chapter.md

- Previously the build would fail due to, e.g., `getaddrinfo` not being defined
- The previous download links were all broken
- Crash when starting on some X11 systems

#### Acceptable backport criteria

The stable branch does have some changes which cannot be backported. Most notable are breaking changes. The desire is to have stable users be uninterrupted when updating packages.

However, many changes are able to be backported, including:
- New Packages / Modules
- Security / Patch updates
- Version updates which include new functionality (but no breaking changes)

> - Software which is always meant to be at it's latest. (E.g. `spotify`, `steam`, or `discord`)

Reviewer: ```suggestion
- Software which is always meant to be the most up-to-date. (E.g. `spotify`, `steam`, or `discord`)
```

- Previously the build would fail due to, e.g., `getaddrinfo` not being defined
- The previous download links were all broken
- Crash when starting on some X11 systems

#### Acceptable backport criteria

> The stable branch does have some changes which cannot be backported. Most notable are breaking changes. The desire is to have stable users be uninterrupted when updating packages.

Author: Yea, it's not great
